pkg/node/manager/manager.go

```diff
@@ -439,22 +440,10 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 		}
 
 		ipAddrStr := address.IP.String()
-		_, err := m.ipcache.Upsert(ipAddrStr, tunnelIP, key, nil, ipcache.Identity{
-			ID:     remoteHostIdentity,
-			Source: n.Source,
-		})
-
+		m.ipcache.UpsertAuxiliary(ipAddrStr, tunnelIP, key)
 		m.upsertIntoIDMD(ipAddrStr, remoteHostIdentity)
 
-		// Upsert() will return true if the ipcache entry is owned by
-		// the source of the node update that triggered this node
-		// update (kvstore, k8s, ...) The datapath is only updated if
-		// that source of truth is updated.
-		if err != nil {
-			dpUpdate = false
```

Comment:
I believe this is somewhat safe to remove, as we also have a source overwrite check here:
https://github.com/cilium/cilium/blob/f9179b3b351d55ee54a5fa9538751dcac44b17ca/pkg/node/manager/manager.go#L470-L477

Reply:
I presume you're referring to the removal of the `dpUpdate` logic?

From what I understand, the idea is that we first try to inject an ipcache entry, which could plausibly fail (for instance if the node update is for the local node, where we already knew about the local node locally via `NodeDiscovery`), so then in that case we don't need to go and update the datapath and invoke `nh.NodeUpdate(...)` / `nh.NodeAdd(...)` below. 

So I think the result is that all of the NodeManager subscribers (hubble and datapath, where datapath does the route updates, tunnel map updates, ipsec/wireguard config etc.) will now receive updates even if the ipcache suggests that nothing needs to change. Hopefully those subscribers are mostly idempotent anyway?

IMHO the previous logic where `ipsAdded` was extended or not based on the success of the ipcache modification seemed a bit odd, that doesn't change whether the node's IP is changing or not. :shrug: